thread/thread.cpp

```diff
@@ -2095,6 +2093,10 @@ R"(
     }
 
     int vcpu_init(uint64_t flags) {
+        if (unlikely(!photon_thread_alloc || !photon_thread_dealloc)) {
```

Comment:
The only problem is here:
once vcpu_init called in a global constructor, it might be called before `photon_thread_alloc` construct or library load.

that may cause delegate `photon_thread_alloc` become an empty delegate and unable to be called, result in allocation failure and segment fault.